zebra-chain/src/transaction/lock_time.rs

```diff
@@ -82,10 +102,10 @@ impl ZcashSerialize for LockTime {
 impl ZcashDeserialize for LockTime {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
         let n = reader.read_u32::<LittleEndian>()?;
-        if n <= block::Height::MAX.0 {
+        if n < Self::MIN_TIMESTAMP as u32 {
```

Comment:
Risky Code: I try to avoid `as`, because it silently truncates values.

(It's ok in constant definitions, because `Into` and `TryInto` aren't `const`. And we'll stop using i64 when we switch to `DateTime32` in #2211.)

```suggestion
        if n < Self::MIN_TIMESTAMP.try_into().expect(msg: "fits in u32") {
```